feat(cargo-wdk): add `--signtool-args` passthrough to customize driver signing (#699)
microsoft/windows-drivers-rs: crates/cargo-wdk/README.md

`--verify-signature` cannot be combined with `--sign-mode=off` because if signing is off there is nothing to verify. Passing both will cause `build` to fail with an error.

##### Customizing signtool arguments

To sign with your own certificate or tweak any signing option, pass `--signtool-args` with a string of the arguments to forward to `signtool sign`.

- When `--signtool-args` is **omitted**, cargo-wdk signs with the auto-generated WDR test certificate as described above.
- When `--signtool-args` is **provided**, you own the full `signtool sign` option set (certificate selection, digest algorithm, etc.). cargo-wdk only wraps your arguments with the `sign` verb and the trailing file operand. If your arguments don't include a certificate selector (e.g. `/n`, `/s`, `/f`, `/sha1`), `signtool` auto-selects a code-signing certificate from your personal store and fails if none — or more than one — is found.

> `--signtool-args` applies only when signing is enabled; supplying it with `--sign-mode=off` is an error.

Author: Ideally additional files must not be passed via `signtool-args`. No need to add additional details since the packaging flow is known.

`--verify-signature` cannot be combined with `--sign-mode=off` because if signing is off there is nothing to verify. Passing both will cause `build` to fail with an error.

##### Customizing signtool arguments

To sign with your own certificate or tweak any signing option, pass `--signtool-args` with a string of the arguments to forward to `signtool sign`.

- When `--signtool-args` is **omitted**, cargo-wdk signs with the auto-generated WDR test certificate as described above.

> - When `--signtool-args` is **provided**, you own the full `signtool sign` option set (certificate selection, digest algorithm, etc.). cargo-wdk only wraps your arguments with the `sign` verb and the trailing file operand. If your arguments don't include a certificate selector (e.g. `/n`, `/s`, `/f`, `/sha1`), `signtool` auto-selects a code-signing certificate from your personal store and fails if none — or more than one — is found.

Reviewer: ```suggestion
- When `--signtool-args` is **provided**, you own the full `signtool sign` option set (certificate selection, digest algorithm, etc.). `cargo-wdk` will prepend the `sign` verb to your arguments and append the trailing file operand so you should not provide them. If your arguments don't include a certificate selector (e.g. `/n`, `/s`, `/f`, `/sha1`), `signtool` auto-selects a code-signing certificate from your personal store and fails if none, or more than one, is found.
```

Author: Applied the suggested changes ✅

    ```pwsh
    cargo wdk build --sign-mode off
    ```

- To test-sign with a certificate selected from a store by subject name, run:

    ```pwsh
    cargo wdk build --signtool-args "/s MyStore /n MyCert /fd SHA256"
    ```

- To test-sign with a PFX file whose password is supplied inline, run:

    ```pwsh
    cargo wdk build --signtool-args "/f C:\certs\my.pfx /p <password> /fd SHA256"
    ```

>     > **Warning:** Passing a password inline via `/p` exposes it to other processes on the machine (e.g. through `Get-Process`/task managers) and risks it ending up in logs. cargo-wdk omits the signtool arguments from its own command logging, but the shell history and process arguments are outside its control. Prefer importing the PFX into a certificate store and selecting it with `/n`/`/sha1`, or provide the password through a more secure mechanism, instead of putting it on the command line.

Reviewer: This warning is not needed since we're redacting the password in logs. Please remove it.

>     > **Warning:** Passing a password inline via `/p` exposes it to other processes on the machine (e.g. through `Get-Process`/task managers) and risks it ending up in logs. cargo-wdk omits the signtool arguments from its own command logging, but the shell history and process arguments are outside its control. Prefer importing the PFX into a certificate store and selecting it with `/n`/`/sha1`, or provide the password through a more secure mechanism, instead of putting it on the command line.

Author: Removed ✅

    ```pwsh
    cargo wdk build --sign-mode off
    ```

- To test-sign with a certificate selected from a store by subject name, run:

    ```pwsh
    cargo wdk build --signtool-args "/s MyStore /n MyCert /fd SHA256"
    ```

- To test-sign with a PFX file whose password is supplied inline, run:

    ```pwsh
    cargo wdk build --signtool-args "/f C:\certs\my.pfx /p <password> /fd SHA256"

>     ```

Reviewer: Don't add these examples. Like I said in the past, we don't have to add examples for every capability we have.

Author: Removed ✅

    ```pwsh
    cargo wdk build --sign-mode off
    ```

- To test-sign with a certificate selected from a store by subject name, run:

    ```pwsh
    cargo wdk build --signtool-args "/s MyStore /n MyCert /fd SHA256"
    ```

- To test-sign with a PFX file whose password is supplied inline, run:

    ```pwsh
    cargo wdk build --signtool-args "/f C:\certs\my.pfx /p <password> /fd SHA256"

>     ```

Author: Added a note regarding the same
